src/workers/_inputs.js

```diff
@@ -1,348 +1,110 @@
+    let best_fields = result?.best_fields;
 
-/**************************
- ****** HDF5 loaders ******
- **************************/
-
-function load10XRaw(files) {
-    utils.freeCache(cache.matrix);
-
-    // In theory, we could support multiple HDF5 buffers.
-    var first_file = files[0];
-    var tmppath = first_file.name;
-    scran.writeFile(tmppath, new Uint8Array(first_file.buffer));
-
-    try {
-        cache.matrix = scran.initializeSparseMatrixFromHDF5(tmppath, "matrix");
-
-        // Fetching the gene IDs and names.
-        cache.genes = null;
-        let objects = scran.extractHDF5ObjectNames(tmppath);
-        if ("features" in objects["matrix"]) {
-            let fobjects = objects["matrix"]["features"];
-            if ("id" in fobjects && fobjects["id"] === "string dataset") {
-                cache.genes = { id: scran.loadHDF5Dataset(tmppath, "matrix/features/id").contents };
-                if ("name" in fobjects && fobjects["name"] === "string dataset") {
-                    cache.genes.names = scran.loadHDF5Dataset(tmppath, "matrix/features/name").contents;
-                }
-            }
-        }
-
-        // TODO: pull out sample IDs from the HDF5 file, if they exist.
-        cache.annotations = null;
-
-    } finally {
-        scran.removeFile(tmppath);
+    let gnames = [], mats = [];
+    for (var i = 0; i < keys.length; i++) {
+        gnames.push(datasets[keys[i]].genes[best_fields[i]]);
+        mats.push(datasets[keys[i]].matrix);
     }
 
-    if (cache.genes === null) {
-        cache.genes = dummyGenes(cache.matrix.numberOfRows());
-    }
-    scran.permuteFeatures(cache.matrix, cache.genes);
-
-    return;
-}
-
-function loadH5ADRaw(files, name) {
-    utils.freeCache(cache.matrix);
-
-    // In theory, we could support multiple HDF5 buffers.
-    var first_file = files[0];
-    var tmppath = first_file.name;
-    scran.writeFile(tmppath, new Uint8Array(first_file.buffer));
-
-    try {
-        cache.matrix = scran.initializeSparseMatrixFromHDF5(tmppath, "X");
-        let objects = scran.extractHDF5ObjectNames(tmppath);
-
-        // Trying to guess the gene names.
-        cache.genes = null;
-        if ("var" in objects) {
-            let vobjects = objects["var"];
-            if (utils.isObject(vobjects)) {
-                if ("_index" in vobjects && vobjects["_index"] == "string dataset") {
-                    cache.genes = { "_index": scran.loadHDF5Dataset(tmppath, "var/_index").contents };
-                    for (const [key, val] of Object.entries(vobjects)) {
-                        if (val === "string dataset" && (key.match(/name/i) || key.match(/symb/i))) {
-                            cache.genes[key] = scran.loadHDF5Dataset(tmppath, `var/${key}`).contents;
-                        }
-                    }
-                }
-            }
-        }
+    // cbind assays with names
+    let merged_mats = scran.cbindWithNames(mats, gnames);
```

Comment:
Don't do a `cbind` if there is only one dataset.

Reply:
it doesn't L24 would catch this